rails-tasks/app/controllers/tasks_controller.rb

```diff
@@ -29,11 +31,33 @@ def create
   end
 
   def destroy
-    to_delete = Task.where(task_params[:name])
-    Task.destroy(to_delete[0].id)
+    to_delete = Task.find_by(name: params[:name])
+    Task.destroy(to_delete.id)
     redirect_to "/"
   end
 
+  def complete
+    task = Task.find_by(name: "#{params[:name]}")
+    task.update(completed_date: "#{Time.now}", complete: true)
+    @task_name = task.name
+    @description = task.description
+    date = task.completed_date
+    @completed_date = "#{date.month}/#{date.day}/#{date.year}"
+  end
+
+  def edit
+    @task = Task.find_by(name: params[:name])
+    @url = '/tasks/:name/update'
```

Comment:
For this variable I would recommend using a hash like `{controller: 'tasks', action: 'update'}` rather than the route's URL string.